jetson/face_extractor.py

```diff
@@ -1,56 +1,31 @@
 import argparse
 from glob import glob
-import hashlib
 import os
 import warnings
 
 import cv2
 import numpy as np
 import torch
-from PIL import Image
 from tqdm import tqdm
 
-from face_detector import FaceDetector
+from face_detector_threaded import FaceDetector
 
 warnings.filterwarnings('once')
 
 
-# for videos, saves filenames and hashes
-# for images, saves filenames
-def get_files(input_dir, hash_file=None, images=False):
-    if images:
-        img_ext = ['.jpg', '.JPG', '.png', '.PNG']
-        imgs = [glob(f"{input_dir}/*{e}", recursive=args.recursive) for e in img_ext]
-        images = [img for subimg in imgs for img in subimg]
-        return images
-    else:
-        video_ext = ['.mp4', '.mov', '.MOV', '.MP4']
-        videos = [glob(f"{input_dir}/*{e}", recursive=args.recursive) for e in video_ext]
-        # Reduce the 2d list from before to a 1d list
-        videos = [vid for subvid in videos for vid in subvid]
-
-        hashes = []
-        for video in videos:
-            hasher = hashlib.md5()
-            buf_size = 65536
-            with open(video, 'rb') as video_file:
-                buf = video_file.read(buf_size)
-                while len(buf) > 0:
-                    hasher.update(buf)
-                    buf = video_file.read(buf_size)
-            hashes.append(hasher.hexdigest())
-
-        if hash_file is None:
-            return list(zip(videos, hashes))
-        else:
-            with open(hash_file, 'r') as f:
-                processed_hashes = [line.strip() for line in f]
-            # Filter out the videos that have already been processed
-            return [(file, file_hash) for (file, file_hash) in zip(videos, hashes) if file_hash not in processed_hashes]
+def get_files(input_dir, images):
```

Comment:
can you [type hint](https://www.python.org/dev/peps/pep-0484/#type-definition-syntax) this function definition and provide some documentation on these parameters. After reading the function implementation it was clear how it worked but the function signature itself failed to make that clear. 

Also this function might be trying to do too much. I'd much rather see 2 functions `get_images(input_dir: str)` and `get_videos(input_dir: str)`. This is for 2 reasons, code readability and maintainability. These function names are more descriptive which makes them easier to read and if formats need to be added or removed its much clearer where code needs to be changed

Reply:
input_dir and images are from their respective args. Should I just use the global args.input_dir and args.images?

Reply:
kinda confused at what you're asking here. can you elaborate more

Reply:
My newest commit implements it. I used `args.input_dir` directly in `get_images` and `get_videos`. Is that good practice or should I pass `args.input_dir` as a parameter of `get_images` and `get_videos`?